src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/workflow/GerritReviewStep.java

```diff
@@ -0,0 +1,144 @@
+import org.jenkinsci.plugins.workflow.steps.StepContextParameter;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+/**
+ * Allows altering the Gerrit review posted at the end of build during the build.
+ * @author Teemu Murtola &lt;teemu.murtola@gmail.com&gt
+ */
+public class GerritReviewStep extends AbstractStepImpl {
+
+    private String customUrl;
+    private String unsuccessfulMessage;
+
+    /**
+     * Constructor.
+     *
+     * There are no mandatory parameters to the step.
+     */
+    @DataBoundConstructor
+    public GerritReviewStep() {
+    }
+
+    /**
+     * Gets the custom URL for a step.
+     * @return the URL.
+     */
+    @CheckForNull
+    public String getCustomUrl() {
+        return customUrl;
+    }
+
+    /**
+     * Sets a custom URL to post for a build.
+     * @param customUrl the URL to post.
+     */
+    @DataBoundSetter
+    public void setCustomUrl(String customUrl) {
+        this.customUrl = Util.fixEmptyAndTrim(customUrl);
+    }
+
+    /**
+     * Gets the unsuccessful message for a step.
+     * @return the message.
+     */
+    @CheckForNull
+    public String getUnsuccessfulMessage() {
+        return unsuccessfulMessage;
+    }
+
+    /**
+     * Sets additional information to post for a failed build.
+     * @param unsuccessfulMessage Additional information to post for a failed build.
+     */
+    @DataBoundSetter
+    public void setUnsuccessfulMessage(String unsuccessfulMessage) {
+        this.unsuccessfulMessage = Util.fixEmptyAndTrim(unsuccessfulMessage);
+    }
+
+    /**
+     * Executes the GerritReviewStep.
+     */
+    public static class Execution extends AbstractSynchronousStepExecution<Void> {
+
+        @StepContextParameter
+        private transient Run build;
+
+        @Inject
+        private transient GerritReviewStep step;
+
+        @Override
+        protected Void run() throws Exception {
+            ToGerritRunListener listener = ToGerritRunListener.getInstance();
+            String customUrl = step.getCustomUrl();
+            if (customUrl != null) {
+                listener.setBuildCustomUrl(build, customUrl);
+            }
+            String unsuccessfulMessage = step.getUnsuccessfulMessage();
+            if (unsuccessfulMessage != null) {
+                listener.setBuildFailureMessage(build, unsuccessfulMessage);
+            }
+            return null;
+        }
+
+        private static final long serialVersionUID = 1L;
+    }
+
+    /**
+     * Adds the step as a workflow extension.
+     */
+    @Extension(optional = true)
+    public static class DescriptorImpl extends AbstractStepDescriptorImpl {
+
+        /**
+         * Constructor.
+         */
+        public DescriptorImpl() {
+            super(Execution.class);
+        }
+
+        @Override
+        public String getFunctionName() {
+            return "gerritReview";
```

Comment:
I thought about the alternatives a bit, but I do not have a strong preference, so I can change this to something else as well. gerritSetReview or setGerritReview are reasonable alternatives. The reason why I ended up with this name is that I thought that in the long run, all the below four alternatives could be reasonably useful for someone:

* Change the final comment/vote for the change that triggered this build, as done here.
* Post comments/votes to other changes, once this build and/or the triggered builds finish.
* Post immediately to the change that triggered this build.
* Post immediately to some other change.

And for all of these, the parameters influencing what to post would be essentially identical. So the same step could do all of them, with just a separate parameter (e.g., postImmediately: true) deciding how to post, and the name without "set" is suitable for all. But if you prefer, I'm fine with having separate names for these different use cases. Because of backwards-compatibility concerns, it can be difficult to change from one approach to another, though.